sdks/cpp/lite/src/Param.cpp

```diff
@@ -22,6 +22,7 @@ void Param<int32_t>::toProto(Value& value) const {
 
 template <>
 void Param<int32_t>::fromProto(const Value& value) {
+    if (read_only_) throw std::runtime_error("Cannot set read-only parameter");
```

Comment:
it's better to not throw an exception here, just don't call the following `fromProto()` and return from the function

Reply:
same comment for the other `throw` statements in this file